### PR TITLE
tasks: Fix task not executed correctly

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -44,8 +44,7 @@ clean:
 
 test:
 	@echo "run test"
-	@go test -race -coverprofile=coverage.txt -covermode=atomic -v ./...
-	@go tool cover -html="coverage.txt" -o "coverage.html"
+	@go test -race -v -count=1 ./task
```
Comment (Prnyself) [inline Makefile:47]: what does this count flag mean?

Comment (Xuanwo) [reply Makefile:47]: It's a trick to let `go test` ignore cache.

File: proto/staff.proto [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    string leader_id = 3;
+    repeated string worker_ids = 4;
+}
+
+message ClockinRequest {
+    string staff_id = 1;
+    string task_id = 2;
+}
+
+message ClockinReply {
```
Comment (Prnyself) [inline proto/staff.proto:40]: remove blank line?

Comment (Xuanwo) [reply proto/staff.proto:40]: IDEA format proto file like this, maybe we can use an extrenal tool to do this.

Comment (Xuanwo) [reply proto/staff.proto:40]: Maybe https://github.com/uber/prototool#prototool-format ?

Comment (Xuanwo) [reply proto/staff.proto:40]: Or https://github.com/bufbuild/buf

File: task/leader.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+	l.workerClock.Wait()
+	return
+}
+
+func (l *Leader) Handle(ctx context.Context, job *proto.Job) (err error) {
+	l.clockinWorkers()
+
+	reply := &proto.JobReply{}
+	err = l.queue.RequestWithContext(ctx, l.subject, job, reply)
```
Comment (Prnyself) [inline task/leader.go:113]: `reply` is not used here for now?

Comment (Xuanwo) [reply task/leader.go:113]: Yes, we just `JobReply` as an ack for now.

File: task/leader.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	if err != nil {
+		l.logger.Error("leader request",
+			zap.String("subject", l.subject),
+			zap.Error(err))
+		return
+	}
+
+	l.clockoutWorkers()
```
Comment (Prnyself) [inline task/leader.go:121]: why not use `defer` here?

File: task/staff.go [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+	} else {
+		err = HandleAsWorker(s.ctx, electReply.Addr, electReply.Subject, storages)
+	}
+	if err == nil {
+		tr.Status = JobStatusSucceed
+	} else {
+		tr.Status = JobStatusFailed
+		tr.Message = fmt.Sprintf("task handle: %v", err)
+	}
+	err = nil
```
Comment (Prnyself) [inline task/staff.go:162]: this can be removed?

Comment (Xuanwo) [reply task/staff.go:162]: No, we do need to set `err` to nil here.

**Review by Prnyself — APPROVED** [review]

**Pull request merged**